nomos-services/data-availability/sampling/src/lib.rs

```diff
@@ -301,163 +323,143 @@ where
+    SamplingStorage: DaStorageAdapter<Blob = DaBlob> + Send + Sync,
+    VerifierBackend:
+        nomos_da_verifier::backend::VerifierBackend<DaBlob = SamplingBackend::Blob> + Send,
+    VerifierBackend::Settings: Clone,
+    VerifierNetwork: nomos_da_verifier::network::NetworkAdapter + Send,
+    VerifierNetwork::Settings: Clone,
+    VerifierStorage: nomos_da_verifier::storage::DaStorageAdapter + Send,
+    ApiAdapter: ApiAdapterTrait<
+            Blob = SamplingBackend::Blob,
+            BlobId = SamplingBackend::BlobId,
+            Commitments = SamplingBackend::SharedCommitments,
         > + Send
-        + Sync
-        + 'static,
-    Backend::Settings: Clone + Send + Sync + 'static,
-    ApiAdapter: api::ApiAdapter<
-            Blob = Backend::Blob,
-            BlobId = Backend::BlobId,
-            Commitments = Backend::SharedCommitments,
-        > + Send
-        + Sync
-        + 'static,
-    ApiAdapter::Settings: Clone + Send + Sync + 'static,
-    DaNetwork: NetworkAdapter + Send + Sync + 'static,
-    DaNetwork::Settings: Clone + Send + Sync + 'static,
-    DaStorage: DaStorageAdapter<Blob = Backend::Blob> + Sync + Send,
-    DaVerifierBackend:
-        nomos_da_verifier::backend::VerifierBackend<DaBlob = Backend::Blob> + Send + Sync + 'static,
-    DaVerifierBackend::Settings: Clone,
-    DaVerifierNetwork: nomos_da_verifier::network::NetworkAdapter,
-    DaVerifierNetwork::Settings: Clone,
-    DaVerifierStorage: nomos_da_verifier::storage::DaStorageAdapter + Send + Sync,
+        + Sync,
+    ApiAdapter::Settings: Clone + Send + Sync,
 {
     fn init(
         service_state: OpaqueServiceStateHandle<Self>,
         _init_state: Self::State,
     ) -> Result<Self, DynError> {
-        let DaSamplingServiceSettings {
-            sampling_settings,
-            api_adapter_settings: api_backend_settings,
-            ..
-        } = service_state.settings_reader.get_updated_settings();
-
-        let network_relay = service_state.overwatch_handle.relay();
-        let storage_relay = service_state.overwatch_handle.relay();
-        let verifier_relay = service_state.overwatch_handle.relay();
-        let rng = SamplingRng::from_entropy();
-
-        Ok(Self {
-            network_relay,
-            storage_relay,
-            verifier_relay,
-            service_state,
-            sampler: Backend::new(sampling_settings, rng),
-            api_adapter: ApiAdapter::new(api_backend_settings),
-        })
+        Ok(Self::new(service_state))
     }
 
-    async fn run(self) -> Result<(), DynError> {
+    async fn run(mut self) -> Result<(), DynError> {
         // This service will likely have to be modified later on.
         // Most probably the verifier itself need to be constructed/update for every
         // message with an updated list of the available nodes list, as it needs
         // his own index coming from the position of his bls public key landing
         // in the above-mentioned list.
```

Comment:
Now that I see this comment, is not needed anymore.

Reply:
Removed in https://github.com/logos-co/nomos/pull/1132/commits/79c4d0f18b8d7db6c21505339a401f07b39c1940.